### PR TITLE
HIVE-21634

### PR DESCRIPTION
**Pull request by jcamachor**

### CODE REVIEW

File: ql/src/test/queries/clientpositive/perf/mv_query67.q [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+       group by rollup(i_category, i_class, i_brand, i_product_name, d_year, d_qoy, d_moy,s_store_id))dw1) dw2
+where rk <= 100
+order by i_category
+        ,i_class
+        ,i_brand
+        ,i_product_name
+        ,d_year
+        ,d_qoy
+        ,d_moy
+        ,s_store_id
```
Comment (vineetgarg02) [inline ql/src/test/queries/clientpositive/perf/mv_query67.q]: Add a test case (negative) with distinct aggregate?

File: ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveAggregateSplitRule.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    final ImmutableBitSet topAggregateGroupSet = ImmutableBitSet.range(0, bottomAggregateGroupSet.cardinality());
+
+    final Map<Integer, Integer> map = new HashMap<>();
+    bottomAggregateGroupSet.forEach(k -> map.put(k, map.size()));
+    ImmutableList<ImmutableBitSet> topAggregateGroupSets = ImmutableBitSet.ORDERING.immutableSortedCopy(
+        ImmutableBitSet.permute(aggregate.groupSets, map));
+
+    final List<AggregateCall> topAggregateCalls = new ArrayList<>();
+    for (int i = 0; i < aggregate.getAggCallList().size(); i++) {
+      AggregateCall aggregateCall = aggregate.getAggCallList().get(i);
```
Comment (vineetgarg02) [inline ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveAggregateSplitRule.java]: Instead of looping over agg call list again I believe you can do this in the first loop itself where `topAggFuctions` is created.

File: ql/src/test/queries/clientpositive/perf/mv_query67.q (hunk 1 above)
Comment (jcamachor) [reply ql/src/test/queries/clientpositive/perf/mv_query67.q]: I have added it in new commit.

File: ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveAggregateSplitRule.java (hunk 2 above)
Comment (jcamachor) [reply ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveAggregateSplitRule.java]: I was trying to bail out quick, but you are right that creating those objects is not such a huge overhead. I have changed it accordingly.
